### PR TITLE
test(form-field): Add new test for form field

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Back when we re wrote some test to React Testing Library, we didn't get around to FormField, this PR addresses those test
Fixes FormField test in: https://github.com/Workday/canvas-kit/issues/286

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes

### CODE REVIEW

File: modules/form-field/react/spec/FormField.spec.tsx [hunk 1]
```diff
@@ -1,152 +1,109 @@
...
-        .at(0)
-        .prop('id')
-    ).toEqual(hintId);
-    expect(component.find(InputComponent).prop('aria-describedby')).toEqual(hintId);
-
-    component.unmount();
+  describe('when rendered', () => {
+    it('should render a label with text Label', () => {
+      const label = 'Label';
+      const {container} = render(
```
Comment (anicholls) [inline modules/form-field/react/spec/FormField.spec.tsx]: `getByLabelText` is much cleaner. `container` should be a last resort

File: modules/form-field/react/spec/FormField.spec.tsx [hunk 2]
```diff
@@ -1,152 +1,109 @@
...
+        <FormField label={label}>
+          <input type="text" />
+        </FormField>
+      );
+      expect(container.querySelector('label')).toContainHTML('Label');
```
Comment (anicholls) [inline modules/form-field/react/spec/FormField.spec.tsx]: If you've set `'Label'` to a variable above, let's use it.

File: modules/form-field/react/spec/FormField.spec.tsx [hunk 3]
```diff
@@ -1,152 +1,109 @@
...
+    it('should render text below the input component', () => {
+      const label = 'Label';
+      const hintText = 'Helpful text goes here.';
+      const {container} = render(
+        <FormField hintText={hintText} label={label}>
+          <input type="text" />
+        </FormField>
+      );
+
+      expect(container.querySelector('p')).toContainHTML('Helpful text goes here.');
```
Comment (anicholls) [inline modules/form-field/react/spec/FormField.spec.tsx]:
```suggestion
      expect(container.querySelector('p')).toContainHTML(hintText);
```

File: modules/form-field/react/spec/FormField.spec.tsx [hunk 4]
```diff
@@ -1,152 +1,109 @@
...
-        <InputComponent />
-      </FormField>
-    );
-
-    expect(component.find(InputComponent).prop('id')).toEqual(inputId);
-    expect(component.find('Label').prop('htmlFor')).toEqual(inputId);
-
-    component.unmount();
+  describe('when rendered with hint text', () => {
+    it('should render text below the input component', () => {
```
Comment (anicholls) [inline modules/form-field/react/spec/FormField.spec.tsx]: Can you also test the `hintId` here as well?

File: modules/form-field/react/stories/stories.tsx [hunk 5]
```diff
@@ -2,30 +2,95 @@
...
+import {StaticStates} from '@workday/canvas-kit-labs-react-core';
+import {TextInput} from '../../../text-input/react';
 import README from '../README.md';
+import {FormFieldLabelPosition} from '../lib/types';
+
+const FormFieldStates = () => (
+  <StaticStates>
+    <ComponentStatesTable
+      rowProps={[
+        {label: 'Error States', props: {required: false}},
```
Comment (anicholls) [inline modules/form-field/react/stories/stories.tsx]: Why is this row needed? Typo?

```suggestion
```

File: modules/form-field/react/stories/stories.tsx [hunk 6]
```diff
@@ -2,30 +2,95 @@
...
+        error: [
+          {value: undefined, label: 'Default'},
+          {value: FormField.ErrorType.Alert, label: 'Alert'},
+          {value: FormField.ErrorType.Error, label: 'Error'},
+        ],
+      })}
+    >
+      {props => (
+        <FormField {...props} hintText="Helpful text goes here." label="Label">
+          <TextInput></TextInput>
```
Comment (anicholls) [inline modules/form-field/react/stories/stories.tsx]: Nit: You may get an error for this not being `<TextInput />`

File: modules/form-field/react/spec/FormField.spec.tsx (hunk 1 above)
Comment (mannycarrera4) [reply modules/form-field/react/spec/FormField.spec.tsx]: In this case, I'm looking for the `Label` and the text associated with that label, `getByLabelText` will return any element associated with that label, which is why i used `container` instead. I have a talked with @NicholasBoll about this and how it was confusing to me that hat method really grabs elements associated with it and not the node `Label`

File: modules/form-field/react/stories/stories.tsx (hunk 5 above)
Comment (mannycarrera4) [reply modules/form-field/react/stories/stories.tsx]: i guess i wanted to show the form field errors, but it's a bit redundant, will remove!

File: modules/form-field/react/spec/FormField.spec.tsx (hunk 1 above)
Comment (anicholls) [reply modules/form-field/react/spec/FormField.spec.tsx]: Whoops - good call out haha. I think I was thinking of `getByText`.

**Pull request merged**